# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/pygobject/unvendor-capi-compat.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Use our t2 package instead of the one vendored by pygobject

  - NoTag <notag@t2sde.org>

--- pygobject-3.52.3/meson.build.vanilla	2025-07-24 12:28:13.823626312 +0200
+++ pygobject-3.52.3/meson.build	2025-07-24 12:32:05.571737112 +0200
@@ -43,13 +43,13 @@
 
 cc = meson.get_compiler('c')
 
-pythoncapi_compat = subproject('pythoncapi-compat')
-pythoncapi_compat_dep = declare_dependency(
-    include_directories: pythoncapi_compat.get_variable('incdir')
-)
-meson.add_dist_script(
-    python,
-    join_paths(meson.current_source_dir(), 'subprojects', 'packagefiles', 'pythoncapi-compat-dist.py'))
+# pythoncapi_compat = subproject('pythoncapi-compat')
+# pythoncapi_compat_dep = declare_dependency(
+#     include_directories: pythoncapi_compat.get_variable('incdir')
+# )
+# meson.add_dist_script(
+#     python,
+#     join_paths(meson.current_source_dir(), 'subprojects', 'packagefiles', 'pythoncapi-compat-dist.py'))
 
 if not with_pycairo.disabled()
   cairo_dep = dependency('cairo', required: with_pycairo.enabled() and cc.get_id() != 'msvc')
--- pygobject-3.52.3/gi/meson.build.vanilla	2025-07-24 12:32:23.562149640 +0200
+++ pygobject-3.52.3/gi/meson.build	2025-07-24 12:32:57.041417524 +0200
@@ -78,7 +78,7 @@
 configure_file(output : 'config.h', configuration : cdata)
 
 giext = python.extension_module('_gi', sources,
-  dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep, pythoncapi_compat_dep],
+  dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep],
   install: true,
   subdir : 'gi',
   c_args: pyext_c_args + main_c_args
@@ -86,7 +86,7 @@
 
 if cairo_dep.found()
   gicairoext = python.extension_module('_gi_cairo', ['pygi-foreign-cairo.c'],
-    dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep, pycairo_dep, cairo_dep, cairo_gobject_dep, pythoncapi_compat_dep],
+    dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep, pycairo_dep, cairo_dep, cairo_gobject_dep],
     install: true,
     subdir : 'gi',
     c_args: pyext_c_args + main_c_args)
